### PR TITLE
Gene Table Documentation & seedDB sql update

### PR DESCRIPTION
**Pull request by yichaoS**
Documentations for gene table updates
(merged content from https://github.com/cBioPortal/datahub/pull/1347)
### What's changed
- Add `Gene Table 2.0` section in `Release Note` section to describe the latest HGNC update
- Added the list of genes updated/removed/added to HGNC in the `gene-update-list` folder
- Included instructions for updating data files to the `For Developers` section
- Updated `cbioportal/docs/Updating-gene-and-gene_alias-tables.md` under cbioportal repo, PR here:
https://github.com/cBioPortal/cbioportal/pull/8460

### To Discuss:
- should we make a new SQL or should we just have the user update the tables themselves? 
- If we want to build new SQL for download, should we separate releases notes into `schema update` and `gene table content update`? 
- where should the data update script instructions go? Now it's under the `For Developers` section, but this should really be tied with `updating gene tables` section.

### CODE REVIEW

File: seedDB/README.md [hunk 1]
```diff
@@ -4,7 +4,16 @@ These files are MySQL database dump files for seeding a new instance of the cBio
 
 The database schema and cBioPortal release follows different numbering cycles since cBioPortal 1.5.0 and database schema 2.1.0. This means that the version numbers won't be identical. cBioPortal 1.9.0 with database schema 2.4.0 removed PDB annotations from the database.
 
-## Latest seed database
+## Release Notes
+### Latest seed database
+#### Gene Table 2.0
```
Comment (n1zea144) [inline seedDB/README.md:9]: I would follow format of previous seed database entries (see image below).  In the section of "Contents of seed database" I would put the information about genes (in a single bullet with sub-bullets).

image

**Review by n1zea144 — CHANGES_REQUESTED** [review]
I have suggestions for rewording that we can discuss over zoom.

File: seedDB/README.md [hunk 2]
```diff
@@ -4,7 +4,16 @@ These files are MySQL database dump files for seeding a new instance of the cBio
...
+We recently updated our gene tables based on the HGNC [Feb 20, 2021 Download](https://www.genenames.org/download/statistics-and-files/) with small modifications listed below.  
+- To minimize loss of data availability, we maintained certain gene entires that become unavailable in current HGNC as supplment- complete lists [HERE](https://github.com/cBioPortal/datahub/blob/gene_update_doc/seedDB/gene-update-list/gene-supp.md)
+- Updated outdated gene entries - complete list [HERE](https://github.com/cBioPortal/datahub/blob/gene_update_doc/seedDB/gene-update-list/gene-update.md).
+- Removed duplciate symbol <> entrez_ID mapping - complete list [HERE](https://github.com/cBioPortal/datahub/blob/gene_update_doc/seedDB/gene-update-list/gene-removed.md)
+
+> All data files in DATAHUB are also updated accordingly with the gene entries updates. The script/process is described [HERE](https://github.com/cBioPortal/datahub-study-curation-tools/tree/master/gene-table-update/data-file-migration)
```
Comment (n1zea144) [inline seedDB/README.md:15]: Should this go in the News page of cBioPortal?  cc: @ritikakundra

File: seedDB/README.md (hunk 1 above)
Comment (oplantalech) [reply seedDB/README.md:9]: I agree with @n1zea144. Maybe the schema can also be updated to 2.12.8 (current version). All new installations are going to work with the latest cBioPortal version, so it does not make sense to keep an old schema. I think that is the reason why we decided to do the seed database updates based on the schema version.

Comment (yichaoS) [reply seedDB/README.md:9]: @n1zea144 @oplantalech make sense. Just followed the previous format and create a whole new section 2.12.8 with gene table update as a bullet point. (The links to  needs to be updated tho, marked that -- needs further revise once the new mySQL dumps are made)

File: seedDB/README.md [hunk 3]
```diff
@@ -119,5 +128,7 @@ Contents of seed database:
 - Gene lengths retrieved from Gencode Release 25 (mapped to GRCh37)
 - Pfam graphics fetched in September 2016
 
-## For developers
-Updating the seed database for Datahub is described [here](Update-Seed-Database.md).
+
+## For Developers
```
[the remark this answers is not in the document]
Comment (yichaoS) [reply seedDB/README.md:132]: @n1zea144 added a link to this section at the end of the overall description.

File: seedDB/README.md [hunk 4]
```diff
@@ -119,5 +128,7 @@ Contents of seed database:
...
+- Updating the seed database for Datahub is described [HERE](Update-Seed-Database.md).
```
[the remark this answers is not in the document]
Comment (yichaoS) [reply seedDB/README.md:148]: @n1zea144 added another bullet point here with the link to this doc.

File: seedDB/README.md [hunk 5]
```diff
@@ -4,7 +4,32 @@ These files are MySQL database dump files for seeding a new instance of the cBio
...
+
+## Release Notes
+### Latest seed database
+
+#### Seed database schema 2.12.8
+
+This schema is required for cBioPortal release versions:
+- **2.0.0**
+
+**** This part needs revise****
```
Comment (oplantalech) [inline seedDB/README.md:17]: This refers to the cBioPortal version where the schema 2.12.8 was introduced. In this case is version 3.6.0. Therefore, this schema can be used for cBioPortal version 3.6.0 or higher (> 3.6.0).

File: seedDB/README.md [hunk 6]
```diff
@@ -4,7 +4,31 @@ These files are MySQL database dump files for seeding a new instance of the cBio
...
+
+#### Seed database schema 2.12.8
+
+This schema is required for cBioPortal release versions:
+- **2.0.0**
+
+When using a release version **> 2.0.0**, a migration step to a new database schema might be required. The migration process is described [here](https://github.com/cBioPortal/cbioportal/blob/master/docs/Updating-your-cBioPortal-installation.md#running-the-migration-script).
+
+**Schema 2.12.8**: [SQL file with create table statements](https://raw.githubusercontent.com/cBioPortal/cbioportal/v3.6.10/db-scripts/src/main/resources/cgds.sql)<br>
+**Seed database**: [seed-cbioportal_hg19_v2.7.3.sql.gz](https://github.com/cBioPortal/datahub/raw/master/seedDB/seed-cbioportal_hg19_v2.7.3.sql.gz)<br>
```
Comment (mandawilson) [inline seedDB/README.md:20]: I will need to give you the new seed database once we have finalised it.

Comment (mandawilson) [reply seedDB/README.md:20]: And the md5sum for it.

File: seedDB/README.md [hunk 7]
```diff
@@ -4,7 +4,31 @@ These files are MySQL database dump files for seeding a new instance of the cBio
...
+md5sum 85444ce645104dbc00610fc1f15e8c7a
+
+Contents of seed database:
+- Entrez Gene IDs, HGNC symbols and gene aliases updated based on HGNC [Feb 20, 2021 Download](https://www.genenames.org/download/statistics-and-files/) with small modifications listed below.  
+  - To minimize loss of data availability, we maintained certain gene entires that become unavailable in current HGNC as supplment- complete lists [HERE](https://github.com/cBioPortal/datahub/blob/gene_update_doc/seedDB/gene-update-list/gene-supp.md)
```
Comment (mandawilson) [inline seedDB/README.md:25]: -> To minimize loss of data we preserved certain gene entries that are unavailable in the current HGNC in a supplemental file - complete lists
Correcting a few typos and maybe making it clearer?

File: seedDB/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mandawilson) [reply seedDB/README.md:147]: @yichaoS Ben is correct that `Update-Seed-Database.md`, I think you showed me another PR with those changes?

File: seedDB/README.md (hunk 7 above)
Comment (yichaoS) [reply seedDB/README.md]: @mandawilson Thanks for the correction! Fixed.

File: seedDB/README.md (hunk 5 above)
Comment (yichaoS) [reply seedDB/README.md]: @oplantalech Thanks! updated the version here

File: seedDB/README.md (hunk 6 above)
Comment (yichaoS) [reply seedDB/README.md]: Thanks @mandawilson added the sql, updated the link and md5checksum.

**Comment by n1zea144** [comment]
PR merged as “Unexpected low number of genes in seed database: 40872” error is due to conversion to hgnc gene datasource and 40872 is the correct number of genes.

**Pull request merged**